jobs/resource_job.go

```diff
@@ -823,6 +823,15 @@ func ResourceJob() *schema.Resource {
 			if js.isMultiTask() {
 				ctx = context.WithValue(ctx, common.Api, common.API_2_1)
 			}
+			// NodeTypeID cannot be set in jobsAPI.Update if InstancePoolID is specified.
+			// If both are specified, assume InstancePoolID takes precedence and NodeTypeID is only computed.
+			if js.NewCluster != nil && js.NewCluster.InstancePoolID != "" {
+				js.NewCluster.NodeTypeID = ""
+			}
+			// Same for Driver.
+			if js.NewCluster != nil && js.NewCluster.DriverInstancePoolID != "" {
+				js.NewCluster.DriverNodeTypeID = ""
+			}
```

Comment:
it makes sense to make it not only for new clusters, but also for job clusters.  We really have following clusters in job resource:

1. new cluster on resource level (compatibility with single task API 2.0)
2. new cluster on task level
3. job clusters

We need to support all of them.  Also it makes sense to fix it for `databricks_cluster` as well. We also have #2141